### DISCUSSION TITLE
How to set supporting constraint and spring constraint?

### DISCUSSION THREAD

**Discussion by DYLDYLDYL** [idaholab/moose#22903] [Q&A Modules: Solid mechanics] [answered]
Hi!
I want to set an axial supporting constraint at the bottom and an axial spring constraints at the top. 

Thanks in advance for your help!

**Comment by GiudGiud** [comment, marked as answer]
Hello

Supporting constraint is likely just a dirichlet BC on the displacement variable
A spring constraint will be a neumann BC in the momentum equation I think, but there's definitely something in the tensor mechanics module for that

Guillaume

**Reply by hugary1995** [reply]
right, fixture support -> Dirichlet BC, spring support -> Robin BC.

**Reply by DYLDYLDYL** [reply]
Thanks a lot ！
